torch/optim/sparse_adam.py

```diff
@@ -79,6 +79,10 @@ def make_sparse(values):
                 exp_avg, exp_avg_sq = state['exp_avg'], state['exp_avg_sq']
                 beta1, beta2 = group['betas']
 
+                if group['weight_decay'] != 0:
+                    param_values = p.data.sparse_mask(grad)._values()
```

Comment:
this means that you only decay the value you train, which is very very unintuitive and weird.

Reply:
Thanks for your review!
Can you share an idea to improve it?

I made it like this because this optimizer is using sparse gradients and only those portions of gradients  get applied to the parameters as explained in the class docstring.

Reply:
Isn't this how the weight decay is implemented in Adam too?